nfc/NfcCx/Cx/NfcCxESE.cpp

```diff
@@ -0,0 +1,1888 @@
+    _Out_opt_bytecap_(OutputBufferLength) PVOID OutputBuffer,
+    _In_ size_t OutputBufferLength
+    )
+    /*++
+
+    Routine Description:
+
+    This routine dispatches the SmartCard Get Attribute
+
+    Arguments:
+
+    Device - Handle to a framework device object.
+    Request - Handle to a framework request object.
+    InputBuffer - The input buffer
+    InputBufferLength - Length of the input buffer.
+    OutputBufferLength - Length of the output buffer associated with the request.
+    OuptutBuffer - The output buffer
+
+    Return Value:
+
+    NTSTATUS.
+
+    --*/
+{
+    NTSTATUS status = STATUS_NOT_SUPPORTED;
+    DWORD *pdwAttributeId = (DWORD*)InputBuffer;
+
+    UNREFERENCED_PARAMETER(Device);
+    UNREFERENCED_PARAMETER(Request);
+    UNREFERENCED_PARAMETER(InputBufferLength);
+    UNREFERENCED_PARAMETER(OutputBuffer);
+    UNREFERENCED_PARAMETER(OutputBufferLength);
+
+    TRACE_FUNCTION_ENTRY(LEVEL_VERBOSE);
+
+    //
+    // Buffer validated by the validation method
+    //
+    _Analysis_assume_(sizeof(DWORD) <= InputBufferLength);
+
+    if (*pdwAttributeId == SCARD_ATTR_DEVICE_IN_USE) {
+        status = STATUS_SUCCESS;
+    }
+
+    TRACE_FUNCTION_EXIT_NTSTATUS(LEVEL_VERBOSE, status);
+    TRACE_LOG_NTSTATUS_ON_FAILURE(status);
+
+    return status;
+}
+
+
+//
+// Dispatched methods below
+//
+
+NTSTATUS
+NfcCxEmbeddedSEInterfaceDispatchGetAttribute(
+    _In_ WDFDEVICE Device,
+    _In_ WDFREQUEST Request,
+    _In_opt_bytecount_(InputBufferLength) PVOID InputBuffer,
+    _In_ size_t InputBufferLength,
+    _Out_opt_bytecap_(OutputBufferLength) PVOID OutputBuffer,
+    _In_ size_t OutputBufferLength
+    )
+    /*++
+
+    Routine Description:
+
+    This routine dispatches the SmartCard Get Attribute
+
+    Arguments:
+
+    Device - Handle to a framework device object.
+    Request - Handle to a framework request object.
+    InputBuffer - The input buffer
+    InputBufferLength - Length of the input buffer.
+    OutputBufferLength - Length of the output buffer associated with the request.
+    OuptutBuffer - The output buffer
+
+    Return Value:
+
+    NTSTATUS.
+
+    --*/
+{
+    NTSTATUS status = STATUS_NOT_SUPPORTED;
+    DWORD *pdwAttributeId = (DWORD*)InputBuffer;
+    size_t cbOutputBuffer = OutputBufferLength;
+    PNFCCX_SC_INTERFACE scInterface;
+
+
+    UNREFERENCED_PARAMETER(Request);
+    UNREFERENCED_PARAMETER(InputBufferLength);
+
+    TRACE_FUNCTION_ENTRY(LEVEL_VERBOSE);
+
+    //
+    // Buffer validated by the validation method
+    //
+    _Analysis_assume_(sizeof(BYTE) <= OutputBufferLength);
+
+    scInterface = NfcCxFdoGetContext(Device)->SCInterface;
+
+    for (USHORT TableEntry = 0; TableEntry < ARRAYSIZE(g_EmbeddedSEAttributeDispatch); TableEntry++) {
+        if (g_EmbeddedSEAttributeDispatch[TableEntry].dwAttributeId == *pdwAttributeId) {
+
+            if (g_EmbeddedSEAttributeDispatch[TableEntry].fRequireLock) {
+                WdfWaitLockAcquire(scInterface->SmartCardLock, NULL);
```

Comment:
I don't think this Lock is needed. It was deprecated at some point and removed from NfcCxSE.cpp. Can it be avoided here as well?